pkg/dao/repositories.go

```diff
@@ -5,52 +5,53 @@ import (
 	"gorm.io/gorm"
 )
 
-type PublicRepository struct {
+// Repository internal (non-user facing) representation of a repository
+type Repository struct {
 	UUID     string
 	URL      string
 	Revision string
 }
 
-func GetPublicRepositoryDao(db *gorm.DB) PublicRepositoryDao {
-	return publicRepositoryDaoImpl{
+func GetRepositoryDao(db *gorm.DB) RepositoryDao {
+	return repositoryDaoImpl{
 		db: db,
 	}
 }
 
-type publicRepositoryDaoImpl struct {
+type repositoryDaoImpl struct {
 	db *gorm.DB
 }
 
-func (p publicRepositoryDaoImpl) FetchForUrl(url string) (error, PublicRepository) {
+func (p repositoryDaoImpl) FetchForUrl(url string) (error, Repository) {
 	repo := models.Repository{}
 	result := p.db.Where("URL = ?", url).First(&repo)
 	if result.Error != nil {
-		return result.Error, PublicRepository{}
+		return result.Error, Repository{}
 	}
-	return nil, PublicRepository{
+	return nil, Repository{
 		UUID:     repo.UUID,
 		URL:      repo.URL,
 		Revision: repo.Revision,
 	}
 }
 
-func (p publicRepositoryDaoImpl) List() (error, []PublicRepository) {
+func (p repositoryDaoImpl) List() (error, []Repository) {
```

Comment:
```suggestion
func (p repositoryDaoImpl) ListPublicRepositories() (error, []Repository) {
```
or maybe `ListPublic()`, since isn't listing all repositories, just the ones with the public flag.

Reply:
left this alone and change it to return all Repositories, since introspectAll now uses it